calendar-scheduling/src/main/java/com/linagora/calendar/scheduling/AlarmTriggerService.java

```diff
@@ -0,0 +1,248 @@
+                DEFAULT_CONCURRENCY)
+            .then();
+    }
+
+    private Mono<Void> cleanup(AlarmEvent alarmEvent) {
+        if (alarmEvent.recurring()) {
+            // If the event is recurring, we need to update the alarm time for the next occurrence
+            return alarmInstantFactory.computeNextAlarmInstant(CalendarUtil.parseIcs(alarmEvent.ics()), Username.fromMailAddress(alarmEvent.recipient()))
+                .map(alarmInstant -> Flux.fromIterable(alarmInstant.recipients())
+                    .map(recipient -> new AlarmEvent(alarmEvent.eventUid(), alarmInstant.alarmTime(), alarmInstant.eventStartTime(),
+                        RECURRING, alarmInstant.recurrenceId().map(DateProperty::getValue), recipient, alarmEvent.ics()))
+                    .flatMap(alarmEventDAO::update)
+                    .then())
+                .orElse(alarmEventDAO.delete(alarmEvent.eventUid(), alarmEvent.recipient()));
+        }
+        return alarmEventDAO.delete(alarmEvent.eventUid(), alarmEvent.recipient());
+    }
+
+    private Mono<Void> sendMail(AlarmEvent alarmEvent, Instant now) {
+        Username recipientUser = Username.fromMailAddress(alarmEvent.recipient());
+        return getUserSettings(recipientUser).flatMap(resolvedSettings -> {
+            Locale locale = resolvedSettings.locale();
+            var model = toPugModel(CalendarUtil.parseIcs(alarmEvent.ics()),
+                alarmEvent.recurrenceId(),
+                locale,
+                Duration.between(now, alarmEvent.eventStartTime()));
+            return Mono.fromCallable(() -> messageGeneratorFactory.forLocalizedFeature(new Language(locale), TEMPLATE_TYPE))
+                .flatMap(messageGenerator -> messageGenerator.generate(recipientUser, maybeSender.asOptional(), model, List.of()))
+                .flatMap(message -> mailSenderFactory.create()
+                    .flatMap(mailSender -> mailSender.send(new Mail(maybeSender, List.of(alarmEvent.recipient()), message))));
+        });
+    }
+
+    private Mono<ResolvedSettings> getUserSettings(Username user) {
+        return settingsBasedResolver.readSavedSettings(sessionProvider.createSession(user))
+            .defaultIfEmpty(ResolvedSettings.DEFAULT)
+            .doOnError(error -> {
+                if (!(error instanceof DomainNotFoundException)) {
+                    LOGGER.error("Error resolving user settings for {}, will use default settings: {}",
+                        user.asString(), ResolvedSettings.DEFAULT, error);
+                }
+            }).onErrorResume(error -> Mono.just(ResolvedSettings.DEFAULT));
+    }
+
+    private Map<String, Object> toPugModel(Calendar calendar,
+                                           Optional<String> maybeRecurrenceId,
+                                           Locale locale,
+                                           Duration duration) {
+        VEvent vEvent = maybeRecurrenceId.map(recurrenceId -> getVEvent(calendar, recurrenceId))
```

Comment:
```suggestion
        VEvent vEvent = maybeRecurrenceId.flatMap(recurrenceId -> getVEvent(calendar, recurrenceId))
```

`getVEvent` should return `Optional<VEvent>` instead of throwing an IllegalStateException, it safer